tests/run-make/short-ice/rmake.rs

```diff
@@ -0,0 +1,34 @@
+// Backtraces in internal compiler errors used to be unbearably long, spanning
+// multiple hundreds of lines. A fix was pushed in #108938, and this test gathers
+// varied metrics on level 1 and full-level backtraces to check that the output
+// was shortened down to an appropriate length.
+// See https://github.com/rust-lang/rust/issues/107910
+
+use run_make_support::rustc;
+use std::env;
+
+fn main() {
+    let mut rust_test_1 =
+        rustc().set_backtrace_level("1").input("src/lib.rs").arg("-Ztreat-err-as-bug=1").run_fail();
+    let mut rust_test_2 = rustc()
+        .set_backtrace_level("full")
+        .input("src/lib.rs")
+        .arg("-Ztreat-err-as-bug=1")
+        .run_fail();
+    let rust_test_log_1 = rust_test_1.stderr_utf8().push_str(&rust_test_1.stdout_utf8()).as_str();
```

Comment:
Discussion: wait I'm confused, how does this not suffer from referencing a temporary?

Reply:
Wait how does this compile?

```rs
rust_test_1
    .stderr_utf8()
    .push_str(&rust_test_1.stdout_utf8()) // <- doesn't this return `()`? how does this method call chain work?
    .as_str()
```

Reply:
EDIT: because I'm dumb and this doesn't compile, lol

Reply:
Hehe, who needs a compiler when I have you?

Thanks for finding this silly mistake!

Reply:
Yeah, please ping review when CI is green or you are quite confident that it will compile or you want to discuss something. CI helps to catch easy mistakes.

I think the current test won't compile because we're calling push_str on an immutable binding.